File: client/src/app/features/publications/data/get-publication-dashboard.response.ts
import { DayOfWeek } from '../../../shared/models/day-of-week.enum';
import { PublicationState } from '../../../shared/models/publication-state.enum';
import { SlotState } from '../../../shared/models/slot-state.enum';
import { SlotWindow } from '../../../shared/models/slot-window.enum';

export interface GetPublicationDashboardResponse {
    state: PublicationState;
    windowStartUtc: string | null;
    windowEndUtc: string | null;
    linkToken: string;
    studentsSubmitted: number;
    totalPicks: number;
    lastSubmissionAtUtc: string | null;
    latestExcelVersion: number | null;
    slotCounts: SlotCountForGetPublicationDashboardResponse[];
}

export interface SlotCountForGetPublicationDashboardResponse {
    slotId: string;
    day: DayOfWeek;
    window: SlotWindow;
    state: SlotState;
    requestCount: number;
}

File: src/DrivingLessons.Application/Queries/GetPublicationDashboard/GetPublicationDashboardResponse.cs
using DrivingLessons.Domain.Values;

namespace DrivingLessons.Application.Queries.GetPublicationDashboard;

public class GetPublicationDashboardResponse
{
    public PublicationState State { get; init; }
    public DateTimeOffset? WindowStartUtc { get; init; }
    public DateTimeOffset? WindowEndUtc { get; init; }
    public string LinkToken { get; init; } = string.Empty;
    public int StudentsSubmitted { get; init; }
    public int TotalPicks { get; init; }
    public DateTimeOffset? LastSubmissionAtUtc { get; init; }
    public int? LatestExcelVersion { get; init; }
    public IReadOnlyList<SlotCountForGetPublicationDashboardResponse> SlotCounts { get; init; } = [];
}

public class SlotCountForGetPublicationDashboardResponse
{
    public Guid SlotId { get; init; }
    public DayOfWeek Day { get; init; }
    public SlotWindowType Window { get; init; }
    public SlotState State { get; init; }
    public int RequestCount { get; init; }
}
